iothub/device/src/Message.cs

```diff
@@ -1,39 +1,22 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
 using System.IO;
-using System.Threading;
-using Microsoft.Azure.Devices.Client.Extensions;
-using Microsoft.Azure.Devices.Client.Common.Api;
-using System.Collections.Generic;
 using Microsoft.Azure.Devices.Shared;
 
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
     /// The data structure represent the message that is used for interacting with IotHub.
     /// </summary>
-    public sealed class Message : IReadOnlyIndicator, IDisposable
+    public sealed class Message : MessageBase
```

Comment:
The Message class is split into a `MessageBase`, which has all of the base functionality that is common between both `Message` and `TelemetryMessage`.